File: android-native-app/app/src/main/java/pl/edu/pb/androidnativeapp/data/infinitelist/remote/InfiniteListDataSourceImpl.kt
package pl.edu.pb.androidnativeapp.data.infinitelist.remote

import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import pl.edu.pb.androidnativeapp.data.model.Page
import pl.edu.pb.androidnativeapp.data.network.CharacterApi

class InfiniteListDataSourceImpl(
    private val characterApi: CharacterApi,
) : InfiniteListDataSource {

    override suspend fun getCharactersPage(url: String): Page {
        return withContext(Dispatchers.IO) {
            val response = characterApi.getData(url)
            if (response.isSuccessful) {
                response.body()!!
            } else {
                throw RuntimeException(response.message(), null)
            }
        }
    }
}

File: flutter-multiplatform-app/flutter_multiplatform_app/lib/data/infinitylist/remote/infinite_list_data_source_impl.dart
import 'package:flutter_multiplatform_app/data/infinitylist/remote/infinite_list_data_source.dart';
import 'package:flutter_multiplatform_app/data/model/page.dart';
import 'package:flutter_multiplatform_app/data/network/character_api.dart';


class InfiniteListDataSourceImpl implements InfiniteListDataSource {
  InfiniteListDataSourceImpl({required CharacterApi characterApi})
      : _characterApi = characterApi;

  final CharacterApi _characterApi;

  @override
  Future<Page> getCharactersPage(String url) {
    return _characterApi.getData(url);
  }
}
